### ISSUE TITLE
Normalized description does not validate, but unnormalized description does

### ISSUE THREAD

**Issue by leicray** [openvar/variantValidator#473]
**Describe the bug**
This issue might be related to issue https://github.com/openvar/variantValidator/issues/151. Please merge if appropriate.

A user enquired about being unable to validate the description `NC_000007.14:g.33340897_33505645del` and that does definitely fail validation giving the error message

`Unable to validate the submitted variant NC_000007.14:g.33340897_33505645del against the GRCh38 assembly.`

`Please check your submission and re-submit.`

BTW, this variant was discovered by exome sequencing, so the breakpoints are not known.

The starting nucleotide of the deletion range (in the _BBS9_ gene) corresponds to the first nucleotide of exon 11. Similarly, the final nucleotide of the deletion range corresponds to the last nucleotide of exon 20. It is improbable that the deletion has precisely deleted exons 11 to 20 along with the intervening introns, but lets plough on regardless.

The deletion can be described relative to the MANE Select transcript as `NM_198428.3:c.1201_2298+2del` and that description also fails validation with the same error as the genomic description.

However, if the description is written as `NM_198428.3:c.1099_2298del`, _i.e._ not normalized, there are no validation errors and, as expected, VariantValidator does normalize the description.

Something is not working as expected.

**Comment by Peter-J-Freeman** [comment]
OK, this is a time-out issue caused by a time-out rather than a fail by the looks of it

```
import json
import VariantValidator
vval = VariantValidator.Validator()
variant = 'NC_000007.14:g.33340897_33505645del' 
genome_build = 'GRCh38'
select_transcripts = 'all'
transcript_set = 'refseq'
validate = vval.validate(variant, genome_build, select_transcripts, transcript_set)
validation = validate.format_as_dict(with_meta=True)
print(json.dumps(validation, sort_keys=True, indent=4, separators=(',', ': ')))
```

This validates, but takes ages because the genomic deletion is huge ~164kb. There are also multiple transcripts. Such submissions need to go to the batch tool. VV was never designed as a CNV scale tool and may need additional code to handle large variants. 

NM_198428.3:c.1201_2298+2del runs just fine when I select GRCh37 or GRCh38. Just slow. It returns NM_198428.3:c.1201_2298+2del and NP_940820.1:p.?. 

NM_198428.3:c.1099_2298del validates quickly and returns NM_198428.3:c.1099_2298del NP_940820.1:p.(Gln367_Leu766del). 

Are you sure NM_198428.3:c.1099_2298del should normalize to NM_198428.3:c.1201_2298+2del? I suspect that VV would have picked this up if it was correct and made the necessary adjustment.

**Comment by leicray** [comment]
Any variant description containing an underscore (del or delins) could be triaged to determine the length of the deletion. Those with "excessively long" deletions could trigger a warning to the user to submit the variant to the batch tool for validation. The problem might be determining what the cut-off ought to be. However, a set of variant descriptions with increasing deletion lengths could be used to empirically determine a safe limit.

**Comment by Peter-J-Freeman** [comment]
As a rough guess, I think the cutoff is ~100k. I also want to establish the correct interpretation of the variant here when we get chance as well

I am not certain that NM_198428.3:c.1099_2298del should normalize to NM_198428.3:c.1201_2298+2del because VV is not doing it, and it ought to if it were correct. 

If VV is doing this correctly, then it also looks like long variants internal to coding sequence process much more quickly than intronic variants. Not necessarily surprising as there are additional steps to process intonic variants which also happen to be the longer stages of validation i.e. applying normalization. I do at some point want to reduce the number of normalizations VV performs, also the number of unnecessary variant parsing (also slow for long variants) but I have no idea when I will have time for this

**Comment by Peter-J-Freeman** [comment]
Very true. We need to get this issues under way 

https://github.com/openvar/variantValidator/issues/471

Then revisit this question. OK, hopefully return to this next week

**Comment by Peter-J-Freeman** [comment]
@leicray Can you test these now we have added latest version only filtering

**Comment by leicray** [comment]
The changes that have been made do seem to partly fix the the problem.

The normalised transcript variant description `NM_198428.3:c.1201_2298+2del`, which is derived from the original genome variant description, now does validate correctly. Previously, only the un-normalised description `NM_198428.3:c.1099_2298del` would validate.

However the original genome variant description `NC_000007.14:g.33340897_33505645del` still fails to validate:

`Unable to validate the submitted variant NC_000007.14:g.33340897_33505645del against the GRCh38 assembly.`

**Comment by Peter-J-Freeman** [comment]
The validation of NC_000007.14:g.33340897_33505645del does *not* fail :). It times out. 

This is a matter of education not code changes. Huge deletions need to be sent to the batch validator, not the interactive site. I am not sure what else we can do to speed things up at this stage. I have ideas but need time to implement

**Comment by leicray** [comment]
As I have suggested above, the solution might be to triage submitted del and dup genomic variants to determine the length of the variant. The genomic variant deletion example is 164,748 nucleotides in length.

The length of the variant could be calculated and a message displayed to the user to use the batch validator for deletions or duplications longer than, say, 1,000 nucleotides.

**Comment by Peter-J-Freeman** [comment]
It's not that clear cut though. It depends on a lot of factors e.g. numbers of transcripts, orientation of transcripts to genome. Sometimes 200,000 bases will validate in time. Sometimes it wont. 1000 nucleotides would be selling us way short!

I have no idea where to place an arbitrary cut off. 

But it may be useful so we can direct folks to the batch tool if > cutoff

**Comment by leicray** [comment]
For now, it might help end users who submit "large" deletion or duplication variants if they were to see a notice that says:

"The variant description that you have submitted represents a large sequence range. Validation of this description might time-out for one or more reasons and no validation report will be returned. If that happens, please resubmit the variant description to the batch validation tool: [https://variantvalidator.org/service/validate/batch/](https://variantvalidator.org/service/validate/batch/) which better able to handle variant descriptions of this type."

**Comment by Peter-J-Freeman** [comment]
Think this used to happen. Not sure what changed. Its an interface issue not a VV issue. Will see if I can update the interface when I fix gene filtering

**Comment by ifokkema** [comment]
> For now, it might help end users who submit "large" deletion or duplication variants if they were to see a notice that says:
> 
> "The variant description that you have submitted represents a large sequence range. Validation of this description might time-out for one or more reasons and no validation report will be returned. If that happens, please resubmit the variant description to the batch validation tool: https://variantvalidator.org/service/validate/batch/ which better able to handle variant descriptions of this type."

That is related to: https://github.com/openvar/variantValidator/issues/471. :wink:

**Comment by Peter-J-Freeman** [comment]
Yeah. Its still an interface issue though :)

Need to catch timeouts. Somehow! Sure it used to happen on the interactive site but needs to work for the API too

**Comment by ifokkema** [comment]
I agree, also because the script calling the API may itself time out after 30 seconds.

**Comment by Peter-J-Freeman** [comment]
Think will fix this in #471
